tapgarden/caretaker.go

```diff
@@ -906,24 +916,31 @@ func (b *BatchCaretaker) stateStep(currentState BatchState) (BatchState, error)
 				return
 			}
 
-			select {
-			case b.confEvent <- confEvent:
-
-			case <-confCtx.Done():
-				log.Debugf("Skipping TX confirmation, context " +
-					"done")
-
-			case <-b.cfg.CancelReqChan:
-				cancelErr := b.Cancel()
-				if cancelErr == nil {
+			for !confSend {
+				select {
+				case b.confEvent <- confEvent:
+					confSend = true
+
+				case <-confCtx.Done():
+					log.Debugf("Skipping TX confirmation, " +
+						"context done")
+					confSend = true
```

Comment:
Is this line correct? If the context is done, can we really set `confSend` to `true`? Maybe we should add a comment here.